### DISCUSSION TITLE
How to generate my github skyline?

### DISCUSSION THREAD

**Discussion by Thecode764** [community/community#129740] [Other Feature Feedback, Questions, & Ideas] [answered]
Hi. how to generate my own github skyline?

i visited to https://skyline.github.com but my response is **OK**

**Comment by martinwoodward** [comment]
Hey @Thecode764, afraid we started turning the Skyline microsite off this weekend.  We're looking at alternative ways to generate a 3D model of your contribution graph in the future.  Curious - was it the STL file you wanted so you could print it or were you heading there for something else?

**Reply by davefollett** [reply]
What a bummer, I love printing each years stl file. Hope it comes back. 😓

**Reply by kamermans** [reply]
I've written an open-source replacement for the community, enjoy!

https://github.com/kamermans/github-skyline

It fetches your contribution history from the years of your choosing and generates a customizable parametric OpenSCAD model and STL for you :)

**Reply by Thecode764** [reply]
thank you

**Comment by yashksaini-coder** [comment]
Hi @martinwoodward Hope you guys also add the option to download and display it as a gif file too.

**Reply by martinwoodward** [reply]
Interesting. Were you thinking the 3d version animated or just a top down one like the profile page?

**Reply by yashksaini-coder** [reply]
I was hoping for a 3D version just as a gif 🎞️, but if you people have better to offer, I can't wait to see 😍!

**Comment by avikalpg** [comment]
Haha, and I thought that I had wasted all that time building [the open-source alternative](https://github.com/avikalpg/skyline) last year when the site was not working.

Maybe I can get back into it and resolve all the issues that I stopped working on because the site was back up.
(I also request others to try to contribute to the project. I might not have the time to fix everything very soon.)

**Reply by kamermans** [reply]
I've written one too which is working well for me:
https://github.com/kamermans/github-skyline

**Reply by avikalpg** [reply]
Nice, you have written it in GoLang. I have starred your repository as well. It is good that people with different preferences will have the option to contribute or use any one of these implementations.

**Comment by BeshoyWill** [comment]
Very interesting, are you looking for beautiful animations and music as your contributions come to life in 3D.

**Comment by yashksaini-coder** [comment]
Hi @Thecode764 if you are looking for an open source alternative to create Github skyline then may I suggest  
- https://git-skyline.huakun.tech/

It is an open source skyline project.

**Comment by ebndev** [comment, marked as answer]
Hi all! Excited to share that we’ve [released gh-skyline](https://github.com/github/gh-skyline), a GitHub CLI extension that allows you to render your contribution graph as an STL file for 3D printing!

If you used our former GitHub Skyline site, then this concept may be familiar to you! But we’ve sprinkled a few additional features into the CLI as well. Read all about it in [this discussion](https://github.com/orgs/community/discussions/146467) and please share any comments or questions directly in that post. 😄

**Comment by avikalpg** [comment]
The CLI is amazing! Meanwhile, for those who still want to see the skyline on a website, I have made some changes to my [skyline](https://github.com/avikalpg/skyline) project. It is now available at [gitskyline.vercel.app](https://gitskyline.vercel.app).

Changelog:
1. Migrated the project from ReactJS to NextJS (to be able to store my GitHub personal access token securely).
2. Deployed the website on Vercel.
3. Add a proper base to the skyline (with your username and the time period of the skyline).
4. Add options to download STL or GLTF.
5. Add the option to embed the skyline in your page using an iframe.
6. Played with the lighting to add effects like sunlight (based on the time of day) and indoor lights.

**Comment by FrenkenFlores** [comment]
In [gitprint.store](https://gitprint.store/) you can go farther: you can create a model for user, repo and even for a branch. You can edit your model, models can be created for GitLab or GitHub. Download models totally free.
**Made with extra ❤️ by [FrenkenFlores](https://github.com/FrenkenFlores)**
